### PR TITLE
Easy de-duplication for Matches

### PR DESCRIPTION
**Pull request by jorainer**
This PR adds the `SingleMatchParam` param object to simplify a *de-duplication* of matches, i.e. to select for each query element a single match to one target element or to remove all duplicated matches (1:n) for each query element. This allows:

- remove all duplicated matches (i.e. remove all matches for query element that match more than one target element).
- reduce 1:n matches of a query element to a 1:1 match by keeping only the *closest* match, or a *top ranked* match (based on values in target column that can be specified by the user).

### CODE REVIEW

**Review by andreavicini — APPROVED** [review]
Looks very good to me! I just commented on two small things

File: R/Matched.R [hunk 1]
```diff
@@ -23,7 +23,7 @@
 #' objects (including `SummarizedExperiment` or `QFeatures`). For [QFeatures()]
 #' objects matches to only one of the *assays* within the object is supported.
 #'
-#' @section Creation and subsetting:
+#' @section Ceation and general handling:
```
[the remark this answers is not in the document]
Comment (jorainer) [reply R/Matched.R]: Thanks! fixed.

File: R/Matched.R [hunk 2]
```diff
@@ -1226,6 +1275,82 @@ setMethod("filterMatches", c("Matched", "ScoreThresholdParam"),
...
+                        .extract_elements(
+                            .objectToMatch(object@target, object@targetAssay),
+                            object@matches$target_idx,
+                            sub("target_", "", param@column)))
+                    vals <- vals[order(vals[, 3L],
+                                       decreasing = param@decreasing), ,
+                                 drop = FALSE]
+                    keep <- vals[match(unique(object@matches$query_idx),
+                                       vals[, 2L]), 1L]
+                    object@matches <- object@matches[keep, , drop = FALSE]
```
[the remark this answers is not in the document]
Comment (jorainer) [reply R/Matched.R:1346]: I checked again, but my editor still does the same indentation.

File: R/Matched.R [hunk 3]
```diff
@@ -69,6 +90,25 @@
...
+#'       the smallest value for this (these) column(s) is retained. This is
+#'       equivalent to `TopRankedMatchesParam(n = 1L, decreasing = FALSE)`.
+#'     - `duplicates = "top_ranked"`: select the *best ranking* match for each
+#'       query element. Parameter `column` allows to specify the column by
+#'       which matches are ranked (use `targetVariables(object)` or
+#'       `scoreVariables(object)` to list possible columns). Parameter
+#'       `decreasing` allows to define whether the match with the highest
+#'       (`decreasing = TRUE`) or lowest (`decreasing = FALSE`) value in
+#'       `column` for each *query* will be selected.
+#'
```
Comment (philouail) [inline R/Matched.R:111]: Just checking that i understand but if I can to select the match with highest intensity i would input `column = "maxo" ` (i don't remember if that is the column name)  and then `duplicate = "top_ranked"` ?

**Review by philouail — APPROVED** [review]
For me it's all good i just had one question, you can check the comment :)

File: R/Matched.R (hunk 3 above)
Comment (jorainer) [reply R/Matched.R:111]: yes, exactly - or sort of :) if target is e.g. a `chromPeaks` matrix (where "maxo" would be the peak intensity of a chromatographic peak) then using `duplicate = "top_ranked"`, `column = "target_maxo"` and `decreasing = TRUE` would select for each query the chrom peak with the highest value in the `"maxo"` column. The `"target_"` might be confusing here, but that's the convention we added in `MetaboAnnotation`: to access columns of the *target* object in a `Matched` object the column name (of the original object) needs to be prefixed with `"target_"`. That's why it's maybe best to use `targetVariables(object)` first to check how the columns are actually named.

Comment (philouail) [reply R/Matched.R:111]: Ah yes yes i remember that "target_" convention. Cool thanks then I get it !

**Pull request merged**